src/Symfony/Component/String/GraphemeString.php

```diff
@@ -0,0 +1,346 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+use Symfony\Component\String\Exception\ExceptionInterface;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * Represents a string of Unicode grapheme clusters encoded as UTF-8.
+ *
+ * A letter followed by combining characters (accents typically) forms what Unicode defines
+ * as a grapheme cluster: a character as humans mean it in written texts. This class knows
+ * about the concept and won't split a letter apart from its combining accents. It also
+ * ensures all string comparisons happen on their canonically-composed representation,
+ * ignoring e.g. the order in which accents are listed when a letter has many of them.
```

Comment:
Could we illustrate this description with examples? Might make it easier to follow.

Reply:
I think the link just below is enough, this is just a docblock :)